File: Ahorn/entities/MultiCharacterCutscene.jl
module MultiCharacterCutscene

using ..Ahorn, Maple

@mapdef Entity "DesoloZantas/MultiCharacterCutscene" MultiCharacterCutscene(x::Integer, y::Integer, autoStart::Bool=true, cutsceneId::String="CH0_MODINTRO")

const placements = Ahorn.PlacementDict(
    "Mod Intro Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Chapter End Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Madeline Rest Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Nightmare Intro Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Wake Up Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Memorial Darkside" => Ahorn.EntityPlacement(MultiCharacterCutscene),
    "Poem Cutscene" => Ahorn.EntityPlacement(MultiCharacterCutscene)
)

function Ahorn.selection(entity::MultiCharacterCutscene)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::MultiCharacterCutscene, room::Maple.Room)
    Ahorn.drawSprite(ctx, "scenery/memorial/memorial", entity.x, entity.y)
end

# Nodes: min=0, max=inf
# Basic node rendering not implemented in auto-generated plugin

end

File: framework/extracted/Loenn/entities/MultiCharacterCutscene.lua
local multi_character_cutscene = {}

multi_character_cutscene.name = "DesoloZantas/MultiCharacterCutscene"
multi_character_cutscene.depth = 0
multi_character_cutscene.justification = {0.5, 0.5}
multi_character_cutscene.texture = "scenery/memorial/memorial"
multi_character_cutscene.nodeLimits = {0, 0}

multi_character_cutscene.placements = {
    {
        name = "Mod Intro Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH0_MODINTRO",
            autoStart = true
        }
    },
    {
        name = "Chapter End Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH0_END",
            autoStart = false
        }
    },
    {
        name = "Madeline Rest Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH1_ENDMADELINE",
            autoStart = false
        }
    },
    {
        name = "Nightmare Intro Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH2_INTRO",
            autoStart = true
        }
    },
    {
        name = "Wake Up Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH2_WAKEUP",
            autoStart = false
        }
    },
    {
        name = "Memorial Darkside",
        data = {
            cutsceneId = "MEMORIAL_DARKSIDE",
            autoStart = false
        }
    },
    {
        name = "Poem Cutscene",
        data = {
            cutsceneId = "MAGGYHELPER_CH2_POEM",
            autoStart = false
        }
    }
}

multi_character_cutscene.fieldInformation = {
    cutsceneId = {
        fieldType = "string",
        options = {
            "MAGGYHELPER_CH0_MODINTRO",
            "MAGGYHELPER_CH0_END",
            "MAGGYHELPER_CH1_ENDMADELINE",
            "MAGGYHELPER_CH2_INTRO",
            "MAGGYHELPER_CH2_WAKEUP",
            "MEMORIAL_DARKSIDE",
            "MAGGYHELPER_CH2_POEM"
        }
    },
    autoStart = {
        fieldType = "boolean"
    }
}

return multi_character_cutscene